Lib/test/test_pyclbr.py

```diff
@@ -142,6 +142,12 @@ def defined_in(item, module):
                     self.assertHaskey(dict, name, ignore)
 
     def test_easy(self):
+        import importlib.util
+        if getattr(sys.modules["__main__"], "__spec__", None) is None:
+            sys.modules["__main__"].__spec__ = importlib.machinery.ModuleSpec(
+                name="__main__", loader=None, origin="built-in"
+            )
```

Comment:
I'm not very happy with this specific hack. Would it be possible to have some contextmanager decorator that we could apply on the test method so that the module's spec is only changed for the duration of the test?